src/containers/FormikForm/articleFormHooks.ts

```diff
@@ -18,36 +23,107 @@ import { formatErrorMessage } from '../../util/apiHelpers';
 import { queryTopics, updateTopic } from '../../modules/taxonomy';
 import * as articleStatuses from '../../util/constants/ArticleStatus';
 import { isFormikFormDirty } from '../../util/formHelper';
+import {
+  DraftApiType,
+  DraftStatus,
+  DraftStatusTypes,
+  UpdatedDraftApiType,
+} from '../../modules/draft/draftApiInterfaces';
+import {
+  Author,
+  AvailabilityType,
+  ConvertedDraftType,
+  License,
+  RelatedContent,
+  VisualElement,
+} from '../../interfaces';
+import { ApiConceptType } from '../../modules/concept/conceptApiInterfaces';
+import { NewReduxMessage, ReduxMessageError } from '../Messages/messagesSelectors';
 
-const getFilePathsFromHtml = htmlString => {
+const getFilePathsFromHtml = (htmlString: string): string[] => {
   const parsed = new DOMParser().parseFromString(htmlString, 'text/html');
   const fileNodesArr = Array.from(parsed.querySelectorAll('embed[data-resource=file]'));
-  return fileNodesArr.map(e => e.getAttribute('data-path'));
+  const paths = fileNodesArr.map(e => e.getAttribute('data-path'));
+  return paths.filter((x): x is string => x !== null);
 };
 
-const deleteRemovedFiles = async (oldArticleContent, newArticleContent) => {
+const deleteRemovedFiles = async (oldArticleContent: string, newArticleContent: string) => {
   const oldFilePaths = getFilePathsFromHtml(oldArticleContent);
   const newFilePaths = getFilePathsFromHtml(newArticleContent);
 
   const pathsToDelete = oldFilePaths.filter(op => !newFilePaths.some(np => op === np));
   return Promise.all(pathsToDelete.map(path => deleteFile(path)));
 };
 
+export interface ArticleFormikType {
+  id?: number;
+  slatetitle?: Value;
+  content?: Value;
+  introduction?: Value;
+  metaDescription?: Value;
+  agreementId?: number;
+  articleType: string;
+  status?: DraftStatus;
+  creators: Author[];
+  processors: Author[];
+  rightsholders: Author[];
+  language?: string;
+  license?: string;
+  metaImageAlt?: string;
+  metaImageId?: string;
+  notes?: string[];
+  origin?: string;
+  published?: string;
+  revision?: number;
+  supportedLanguages: string[];
+  tags: string[];
+  updatePublished: boolean;
+  updated?: string;
+  visualElementObject?: VisualElement;
+  grepCodes?: string[];
+  conceptIds: ApiConceptType[];
+  availability?: AvailabilityType;
+  relatedContent: (DraftApiType | RelatedContent)[];
+}
+
+type HooksInputObject = {
+  getInitialValues: (article: Partial<ConvertedDraftType>) => ArticleFormikType;
+  article: Partial<ConvertedDraftType>;
+  t: WithTranslation['t'];
+  articleStatus?: DraftStatus;
+  updateArticle: (art: UpdatedDraftApiType) => Promise<ConvertedDraftType>;
+  updateArticleAndStatus?: (input: {
+    updatedArticle: UpdatedDraftApiType;
+    newStatus: DraftStatusTypes;
+    dirty: boolean;
+  }) => Promise<ConvertedDraftType>;
+  licenses: License[];
```

Comment:
```suggestion
  licenses?: License[];
```
Er dette pluss å sette default verdi til `[]` nok til å fjern license og tilhørende redux fra GrepCodesForm?

Reply:
Ja, men da stemmer jeg heller for å flytte `getArticleFromSlate` funksjonene de andre stedene inn i scope hvor vi har `licenses` slik at det er mindre sjanse for at vi ender opp med å glemme å passe inn licenses en gang den faktisk trengs. Lagde en ny commit på det nå, hva tenker du?

Reply:
Ja, det er sikkert bedre. Det ser bra ut for meg nå